storage/src/mmr/verification.rs

```diff
@@ -26,7 +27,7 @@ pub trait Storage<H: CHasher> {
     fn size(&self) -> u64;
 
     /// Return the specified node of the MMR if it exists & hasn't been pruned.
-    fn get_node(&self, position: u64) -> Option<&H::Digest>;
+    fn get_node(&self, position: u64) -> impl Future<Output = Option<&H::Digest>> + Send;
```

Comment:
I think we should return `Result<&H::Digest, Error>` (as non-existence should probably be treated differently than an error?

Reply:
right, fixed.